test/Form_test.js

```diff
@@ -2414,4 +2416,55 @@ describe("Form", () => {
       expect(node.tagName).eql(tagName);
     });
   });
+
+  describe("Nested forms", () => {
+    it("should call provided submit handler with form state", () => {
+      const onSubmit = sandbox.spy();
+      let innerRef;
+
+      class ArrayTemplateWithForm extends React.Component {
+        constructor(props) {
+          super(props);
+          innerRef = createRef();
+        }
+
+        render() {
+          const innerFormProps = {
+            schema: {},
+            onSubmit,
+          };
+
+          return (
+            <Portal>
+              <div className="array" ref={innerRef}>
+                <Form {...innerFormProps}>
+                  <button className="array-form-submit" type="submit">
+                    Submit
+                  </button>
+                </Form>
+              </div>
+            </Portal>
+          );
+        }
+      }
+
+      const outerFormProps = {
+        schema: {
+          type: "array",
+          title: "my list",
+          description: "my description",
+          items: { type: "string" },
+        },
+        formData: ["foo", "bar"],
+        ArrayFieldTemplate: ArrayTemplateWithForm,
+        onSubmit,
+      };
+      createFormComponent(outerFormProps);
+      const arrayForm = innerRef.current.querySelector("form");
+      const arraySubmit = arrayForm.querySelector(".array-form-submit");
+
+      arraySubmit.click();
+      sinon.assert.calledOnce(onSubmit);
```

Comment:
So we can get more specificity on what exactly called `onSubmit` -- can you create two functions, `outerOnSubmit` and `innerOnSubmit`, then test to see if `outerOnSubmit` is never called and `innerOnSubmit` is called?